### DISCUSSION TITLE
Near Infrared or Grayscale data

### DISCUSSION THREAD

**Discussion by yusuf-jkhan1** [junyanz/pytorch-CycleGAN-and-pix2pix#1343] [Q&A] [answered]
Because the models expect a 3 channel image input, is it a reasonable approach to take a single channel image, such as a gray scale image and just stack it on itself twice resulting in a three channel image, where the pixel value for each channel is exactly the same?

I tried this with pix2pix and it didn't seem like my generator loss was going down at all even after 20 epochs. I think even though most of my images were paired rgb -> infrared, there was a slight offset in the frame, and some images just weren't actually paired for some reason. So I'm going to try cyclegan instead, but wanted to make sure this was a reasonable approach or if others had a preferred approach?

**Comment by junyanz** [comment, marked as answer]
You can use `--input_nc 1` or `--output_nc 1` in your experiments. You don't need to replicate the same channel twice. 
If your input and output pairs have a small offset, it might be better to improve the alignment of data and use pix2pix, rather than treating it as unaligned data and using CycleGan.
